sources/web/datalab/server.ts

```diff
@@ -185,6 +186,21 @@ function handleRequest(request: http.ServerRequest,
     return;
   }
 
+  if (path.indexOf('/_killvm') == 0) {
+    if ('POST' != request.method) {
+      return;
+    }
+    try {
+      let vminfo = info.getVmInfo();
+      childProcess.execSync(
+        'gcloud compute instances stop ' + vminfo.VM_NAME + ' --zone ' + vminfo.VM_ZONE,
```

Comment:
This should explicitly set the project rather than assuming that the current project (which the user can change) is the hosting project.